2019-05-29 12:52:11,330 ERROR: Exception on /user/susan [GET] [in /home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py:1780]
Traceback (most recent call last):
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 2311, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1834, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1737, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/_compat.py", line 36, in reraise
    raise value
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1832, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1818, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/home/user/PycharmProjects/coursera/python_projects/microblog/app/routes.py", line 99, in user
    page, app.config['POSTS_PER_PAGE'], False)
KeyError: 'POSTS_PER_PAGE'
2019-05-29 13:48:50,878 INFO: Microblog startup [in /home/user/PycharmProjects/coursera/python_projects/microblog/app/__init__.py:42]
2019-05-29 13:50:02,232 ERROR: Exception on /index [GET] [in /home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py:1780]
Traceback (most recent call last):
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 2311, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1834, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1737, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/_compat.py", line 36, in reraise
    raise value
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1832, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1818, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/home/user/PycharmProjects/coursera/python_projects/microblog/app/routes.py", line 30, in index
    page, app.config['POSTS_PER_PAGE'], False)
KeyError: 'POSTS_PER_PAGE'
2019-05-29 13:52:20,352 INFO: Microblog startup [in /home/user/PycharmProjects/coursera/python_projects/microblog/app/__init__.py:42]
2019-05-29 13:52:21,959 ERROR: Exception on / [GET] [in /home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py:1780]
Traceback (most recent call last):
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 2311, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1834, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1737, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/_compat.py", line 36, in reraise
    raise value
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1832, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1818, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/home/user/PycharmProjects/coursera/python_projects/microblog/app/routes.py", line 30, in index
    page, app.config['POSTS_PER_PAGE'], False)
KeyError: 'POSTS_PER_PAGE'
2019-05-29 13:56:23,187 INFO: Microblog startup [in /home/user/PycharmProjects/coursera/python_projects/microblog/app/__init__.py:42]
2019-05-29 19:41:36,423 INFO: Microblog startup [in /home/user/PycharmProjects/coursera/python_projects/microblog/app/__init__.py:44]
2019-05-29 19:43:57,548 INFO: Microblog startup [in /home/user/PycharmProjects/coursera/python_projects/microblog/app/__init__.py:44]
2019-06-02 16:44:54,811 INFO: Microblog startup [in /home/user/PycharmProjects/coursera/python_projects/microblog/app/__init__.py:44]
2019-06-02 16:45:47,076 INFO: Microblog startup [in /home/user/PycharmProjects/coursera/python_projects/microblog/app/__init__.py:44]
2019-06-02 16:46:32,506 INFO: Microblog startup [in /home/user/PycharmProjects/coursera/python_projects/microblog/app/__init__.py:44]
2019-06-02 16:46:44,848 INFO: Microblog startup [in /home/user/PycharmProjects/coursera/python_projects/microblog/app/__init__.py:44]
2019-06-02 16:47:02,410 ERROR: Exception on /reset_password_request [POST] [in /home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py:1780]
Traceback (most recent call last):
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 2311, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1834, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1737, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/_compat.py", line 36, in reraise
    raise value
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1832, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/app.py", line 1818, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/user/PycharmProjects/coursera/python_projects/microblog/app/routes.py", line 167, in reset_password_request
    send_password_reset_email(user)
  File "/home/user/PycharmProjects/coursera/python_projects/microblog/app/email.py", line 19, in send_password_reset_email
    user=user, token=token),
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/templating.py", line 134, in render_template
    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/jinja2/environment.py", line 869, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/jinja2/environment.py", line 830, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/jinja2/environment.py", line 804, in _load_template
    template = self.loader.load(self, name, globals)
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/jinja2/loaders.py", line 113, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/templating.py", line 58, in get_source
    return self._get_source_fast(environment, template)
  File "/home/user/PycharmProjects/coursera/venv/lib/python3.5/site-packages/flask/templating.py", line 86, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: email/reset_password.txt
2019-06-02 16:52:08,346 INFO: Microblog startup [in /home/user/PycharmProjects/coursera/python_projects/microblog/app/__init__.py:44]
2019-06-02 21:13:36,505 INFO: Microblog startup [in /home/user/PycharmProjects/coursera/python_projects/microblog/app/__init__.py:47]
2019-06-03 15:36:54,275 INFO: Microblog startup [in /home/user/PycharmProjects/coursera/python_projects/microblog/app/__init__.py:47]
2019-06-04 16:24:16,691 INFO: Microblog startup [in /home/user/PycharmProjects/coursera/python_projects/microblog/app/__init__.py:52]
2019-06-04 16:24:29,854 INFO: Microblog startup [in /home/user/PycharmProjects/coursera/python_projects/microblog/app/__init__.py:52]
2019-06-05 16:46:30,117 INFO: Microblog startup [in /home/user/PycharmProjects/coursera/python_projects/microblog/app/__init__.py:52]
